test/testapps/TestContentPackage/MyPrompt.cs

```diff
@@ -1,16 +1,14 @@
-﻿using Microsoft.AspNetCore.Blazor.Browser.Interop;
+using Microsoft.JSInterop;
+using System.Threading.Tasks;
 
 namespace TestContentPackage
 {
     public static class MyPrompt
     {
-        // Keep in sync with the identifier in the .js file
-        const string ShowPromptIdentifier = "TestContentPackage.showPrompt";
-
-        public static string Show(string message)
+        public static Task<string> Show(string message)
         {
-            return RegisteredFunction.Invoke<string>(
-                ShowPromptIdentifier,
+            return JSRuntime.Current.InvokeAsync<string>(
+                "TestContentPackage.showPrompt", // Keep in sync with identifiers in the.js file
```

Comment:
After we merge this PR to `dev` and there is a new build on the dev feed, we need to remember to go back to the BlazorLib project template and make the equivalent change to this. Can't do so until there is a new build on the dev feed with the new APIs in it.

Reply:
Reminder - we need to do this 😆

Reply:
Yep, done in 36dc3368

Awkwardly the new convention of making the JS functions globally-reachable via an expression of the form `window.something.somethingElse.etcetera` means there's no longer a simple way to auto-generate different per-project namespaces matching the project name in JS code. This is because AFAICT [`dotnet new` doesn't have a built-in way of saying you want to inject the "safe" version of the project name](https://github.com/dotnet/templating/issues/402#issuecomment-285605336) in specific places, so we can't use the project name in the JS code because it might contain dots, and our lookup logic interprets dots as property name separators.

Net result is that I've now hard-coded the namespace of the `showPrompt` example function as `exampleJsFunctions`, which I think is adequate to make the point that developers should change it to something that makes sense in their project.

The only way this would be a problem is if people actually start shipping NuGet packages in which the JS functions are still attached to a global called `exampleJsFunctions`, and then trying to use multiple such packages would lead to name clashes. But I think that's extremely unlikely and would obviously be a package authoring error.

Reply:
Yeah, I would not bother with it... A comment in the docs saying that people should change to a meaningful (and safe) name is necessary for sanity...